lib/appoptics_apm/api/logging.rb

```diff
@@ -137,16 +137,16 @@ def log_start(layer, xtrace = nil, opts = {})
             # continue valid incoming xtrace
             # use it for current context, ensuring sample bit is set
             AppOpticsAPM::XTrace.set_sampled(xtrace)
-
-            md = AppOpticsAPM::Metadata.fromString(xtrace)
-            AppOpticsAPM::Context.fromString(xtrace)
-            log_event(layer, :entry, md.createEvent, opts)
+            md = AppOpticsAPM::Context.getMetadata(); #.makeRandom(true)
+            AppOpticsAPM::Context.setMetadata(md)
+            # AppOpticsAPM::Context.fromString(xtrace)
+            log_event(layer, :entry, AppOpticsAPM::Context.createEvent, opts)
           else
             # discard invalid incoming xtrace
             # create a new context, ensuring sample bit set
-            md = AppOpticsAPM::Metadata.makeRandom(true)
```

Comment:
We haven't fully fixed this randomize thing yet. This is possibly repeated in multiple places